python/pyiceberg/catalog/rest.py

```diff
@@ -179,10 +179,9 @@ def __init__(
             name: Name to identify the catalog
             properties: Properties that are passed along to the configuration
         """
-        super().__init__(name, **properties)
-
-        self.uri = self.property("uri")
-        if credential := self.properties.get("credential"):
+        self.properties = properties
```

Comment:
This bypasses the original problem because it doesn't set `self.properties` with a kwargs copy.

Reply:
Ah, thanks. Great catch! The problem here is that the `super().__init__` is only being called once, and it silently ignores the call on line 186 with the updated properties (that has the token).